lib/inferno/entities/test.rb

```diff
@@ -171,6 +173,19 @@ def block(&block)
 
         alias run block
 
+        def num_orphan_tests
+          @@num_orphan_tests += 1
+        end
+
+        def tag(new_tag = nil)
+          @tag = new_tag.to_s unless new_tag.nil?
+          @tag ||= begin
+            prefix = parent ? parent.tag : 'UNK'
+            suffix = (parent ? parent.tests.find_index(self) + 1 : num_orphan_tests).to_s.rjust(3, '0')
```

Comment:
I think we're unlikely to need 3-digits here. You could always check how many tests are on the parent to see if that many are needed.